RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -4,7 +4,50 @@
 //  Copyright © yagom academy. All rights reserved.
 // 
 
-import Foundation
+enum RockScissorsPaperError: Error {
+    case invalidRockScissorsPaper
+    case invalidMukjippa
+}
 
-print("Hello, World!")
+enum RockScissorsPaperType: Int, CaseIterable {
+    case rock = 1
+    case scissors
+    case paper
+}
 
+enum RockScissorsPaperScenario {
+    case userWin
+    case userLose
+    case draw
+}
+
+var computerChoice: RockScissorsPaperType?
+var endFlag = false
+
+func printRockScissorsPaper() {
+    print("가위(1), 바위(2), 보(3)! <종료 : 0> : ", terminator: " ")
+}
+
+func createRandomRockScissorsPaper() -> RockScissorsPaperType? {
+    let choice = RockScissorsPaperType(rawValue: Int.random(in: 1...3))
+    guard let randomChoice = choice else {
+        return nil
+    }
+    return randomChoice
+}
```

Comment:
```suggestion
func createRandomRockScissorsPaper() -> RockScissorsPaperType {
    let choice = RockScissorsPaperType(rawValue: Int.random(in: 1...3))
    guard let randomChoice = choice else {
        return
    }
    return randomChoice
}
```
함수에서 옵셔널을 리턴하는 것보다 기본타입을 리턴하는 것이 더욱 직관적이라고 생각해서 위와 같은 코드로 작성했었습니다. 하지만 guard let 바인딩이 실패 할 때 어떤 값을 줄지 몰라서 옵셔널로 변경했습니다.  더 나아질 수 있는 부분이 있는지 궁금합니다! 그리고 추가적으로 보통의 경우 함수에서 옵셔널값을 리턴하는 것과 기본타입을 리턴하는 것 둘 중 어떤 것이 더 많이 사용하는 방법인지 궁금합니다.

Reply:
우선 저는 옵셔널을 사용할때, 옵셔널이 의미하는 상황에 맞게 사용하는 편입니다! 

위 함수의 바인딩하는 부분에서는 열거형을 `rawValue`로 초기화하지 못했을때 `else`구문을 타게 되는데요,
함수가 가위,바위,보를 리턴하는 역할이라면 `else`구문에서 다시한번 해당 함수를 호출하여 옵셔널을 사용하지 않고 해결할 수 있을 것 같습니다!

> 그리고 추가적으로 보통의 경우 함수에서 옵셔널값을 리턴하는 것과 기본타입을 리턴하는 것 둘 중 어떤 것이 더 많이 사용하는 방법인지 궁금합니다.
이에 대해서도 보통 기본 타입을 리턴하도록 최대한 구현하고, 값이 없을수 있는 상황에 대해서 옵셔널을 리턴하는 편입니다!